src/vs/platform/window/electron-main/window.ts

```diff
@@ -27,6 +27,7 @@ export interface ICodeWindow extends IDisposable {
 	readonly win: BrowserWindow | null; /* `null` after being disposed */
 	readonly config: INativeWindowConfiguration | undefined;
 
+	readonly previousWorkspace?: IWorkspaceIdentifier | ISingleFolderWorkspaceIdentifier;
```

Comment:
@bpasero I have added this property so that I can read the previous workspace and the profile associated and remove the profile if it is transient. Please take a look and let me know if there is a better way to do this.

Reply:
Not clear to me why this has to go into the `window.ts`, can you not associate workspaces and windows by listening to window events in your code and do this? I would only have such a property if it is used in more places tbh.

Reply:
Yes, I could do that. But getting this from `ICodeWindow` is more elegant.

Reply:
It is not super clear to me, just reading the code what the lifecycle is though. Someone else using this could think this is stored and restored across restarts?

Reply:
Ok! Created this debt item https://github.com/microsoft/vscode/issues/159189 to move it to the place where I use it.